src/History/src/History.cpp

```diff
@@ -0,0 +1,109 @@
+////////////////////////////////////////////////////////////////////////////////
+/**
+* @file History.cpp
+* @brief Function definitions for Calculator History
+*/
+////////////////////////////////////////////////////////////////////////////////
+
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+#include "CalculatorMessages.h"
+#include "flatbuffers/flatbuffers.h"
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+#include "History.h"
+#include "history_generated.h"
+
+namespace calculator {
+
+// ***************************************************************************** /
+// ***************** History public methods ************************************ /
+// ***************************************************************************** /
+
+History::History()
+{
+    findHistoryFileAndSetAllHistory();
+    printHistory();
+}
+
+void History::addToHistory(std::string fullResult)
+{
+    allHistory_.clear();
```

Comment:
I don't love the implementation of this function. Every time you add to the history it completely rewrite and loads from the disk, which is a little wasteful.

Don't worry about fixing it. Just wanted to point it out.